app/middleware/middleware_healthcheck.rb

```diff
@@ -5,36 +5,79 @@ def initialize(app)
 
   def call(env)
     if env['PATH_INFO'.freeze] == '/status'.freeze
-      Rails.logger.tagged "middleware health check" do
+      Rails.logger.tagged "middleware database health check" do
         response = [503, {'Content-Type' => 'application/json', 'X-Powered-By' => 'Metadata-Management-Tool', 'X-Version' => Rails.configuration.version}]
 
-        # checks the database health
-        begin
-          db_healthy = ActiveRecord::Migrator.current_version != 0
-        rescue StandardError => e
-          Rails.logger.error "Database error: #{e}"
-          db_healthy = false
+        queryParams = env['QUERY_STRING']
+        check_database = (queryParams == 'checkDatabase')
+        check_launchpad = (queryParams == 'checkLaunchpad')
+        responseOutput = ""
+
+        # checks the database health when /status?checkDatabase is passed
+        if check_database
+          begin
```

Comment:
I think maybe create a couple functions would be helpful to remove redundant code.   maybe have a function called checkDatabase() and checkLaunchpad() and have it return true/false if healthy, than you can use those functions in the different use cases you have.   I think will make the main body the function a bit smaller.

Reply:
Alright this should be good. Glad you said something because I realized lines 41-45 had been modified as well so I put them back to the way they were.